src/options/options.ts

```diff
@@ -48,3 +49,20 @@ export function makeOptions(options: ProvidedCodeGenOptions): CodeGenOptions {
     ...options
   };
 }
+
+/**
+ * Validate that the options have required variables for custom generation.
+ */
+export function validOptions(options: ProvidedCodeGenOptions): void {
```

Comment:
>validOptions [](start = 16, length = 12)

Seems like we dont have any tests for this one today - Can we add that as part of this PR?

Reply:
I am actually not sure what I think of this method, as I think that the default templates should be defined in the options before it's sent to the `transform` https://github.com/mtennoe/swagger-typescript-codegen/blob/master/src/transform/transformToCodeWithMustache.ts#L40-L52

I guess the validation would still be the same, so yes let's write tests for it!

Reply:
I wrote a few new tests e2925fc

Reply:
Great!